### DISCUSSION TITLE
Magic-Link-Message - > setFlashdata

### DISCUSSION THREAD

**Discussion by alfkoblischke** [codeigniter4/shield#767] [Q&A]
In shield you get a magic-link-message after you have sent in your mail address with the form. Works perfect. 
Then I end up with the magic-link-message.
I would very much like to just put a Text in session Flashdata and then display it on one of my "normal" views. 
Does anybody know how to do this?

**Comment by datamweb** [comment]
> I would very much like to just put a Text in session Flashdata and then display it on one of my "normal" views.
> Does anybody know how to do this?

Hi @alfkoblischke,

Do you mean not to display the following page?

![Qlink](https://github.com/codeigniter4/shield/assets/9530214/9aa7f6ed-f02c-4c64-97cd-ef1af9a9422f)

**Reply by alfkoblischke** [reply]
Hi datamweb,
yes exaxtly. I don´t want this message and instead return redirect to an existing page together with a flashdata message.

**Reply by datamweb** [reply]
For this you need to use custom controller for **MagicLinkController**.
Use the explanations [here](https://codeigniter4.github.io/shield/customization/#extending-the-controllers).

Don't forget to deprecate the default route (magic-link) shields.
```php
service('auth')->routes($routes, ['except' => ['magic-link']]);
```
The image below is an example of a custom **MagicLinkController** implementation via(SMS). You can apply any changes in **magic-link** as below.
![Screenshot 2023-07-23 142929](https://github.com/codeigniter4/shield/assets/9530214/ae9e2e66-ec77-43f2-b838-9a6221df4f56)
